src/app-layer-smtp.c

```diff
@@ -1263,8 +1151,9 @@ static int SMTPPreProcessCommands(
     DEBUG_VALIDATE_BUG_ON((state->parser_state & SMTP_PARSER_STATE_COMMAND_DATA_MODE) == 0);
 
     /* fall back to strict line parsing for mime header parsing */
+    // TODOrust4 use consumed API
```

Comment:
Should this be part of the PR ?
This being the conversion of SMTP to use the AppLayerResult::Incomplete

Reply:
I don't understand the question.

Reply:
Should the conversion of SMTP to use `AppLayerResult::Incomplete` be part of this PR that translates mime into rust ?

Reply:
I think not, but I will check... Is this supposed to be a recent change ?

Reply:
`SMTPPreProcessCommands` does not use `AppLayerResult::Incomplete` but `SMTPGetLine` does

So, it is only in the SMTP headers parsing, not in the DATA mode...